phylo/src/likelihood/likelihood_tests.rs

```diff
@@ -294,19 +296,19 @@ fn setup_simple_reversibility() -> Vec<PhyloInfo> {
 }
 
 #[rstest]
-#[case::jc69(DNA(JC69), &[])]
-#[case::k80(DNA(K80), &[])]
-#[case::hky(DNA(HKY), &[0.22, 0.26, 0.33, 0.19, 0.5])]
-#[case::tn93(DNA(TN93), &[0.22, 0.26, 0.33, 0.19, 0.5970915, 0.2940435, 0.00135])]
-#[case::gtr(DNA(GTR), &[0.1, 0.3, 0.4, 0.2, 5.0, 1.0, 1.0, 1.0, 1.0, 5.0])]
-fn simple_dna_likelihood_reversibility(#[case] model_type: ModelType, #[case] params: &[f64]) {
+#[case::jc69(JC69, &[])]
+#[case::k80(K80, &[])]
+#[case::hky(HKY, &[0.22, 0.26, 0.33, 0.19, 0.5])]
+#[case::tn93(TN93, &[0.22, 0.26, 0.33, 0.19, 0.5970915, 0.2940435, 0.00135])]
+#[case::gtr(GTR, &[0.1, 0.3, 0.4, 0.2, 5.0, 1.0, 1.0, 1.0, 1.0, 5.0])]
+fn simple_dna_likelihood_reversibility(#[case] model_type: DNAModelType, #[case] params: &[f64]) {
     let info = setup_simple_reversibility();
     let model = DNASubstModel::new(model_type, params).unwrap();
-    let likelihood = DNALikelihoodCost { info: &info[0] };
-    let likelihood_rerooted = DNALikelihoodCost { info: &info[1] };
+    let likelihood = SubstitutionLikelihoodCost::new(&info[0], &model);
```

Comment:
nit: consider using either named constants, fields, or functions, instead of accessing indices directly, as indices are meaningless to readers (at least to me)

Reply:
some examples:
```rust
let likelihood = SubstitutionLikelihoodCost::new(meaningful_aaa_from(info), &model);
let likelihood_rerooted = SubsctitutionLikelihoodCost::new(meaningfull_bbb_from(info), &model);
```

Reply:
Done.